### PR TITLE
Move bootloader and fastflash code to load_vector_table function

### PR DESCRIPTION
**Pull request by saleemrashid**
Removes duplicate code and uses a more idomatic method of loading the vector table.

### CODE REVIEW

File: util.h [hunk 1]
```diff
@@ -38,4 +41,18 @@ void __attribute__((noreturn)) system_halt(void);
 // reset system
 void __attribute__((noreturn)) system_reset(void);
 
+static inline void __attribute__((noreturn)) load_vector_table(const vector_table_t *vector_table) {
+    // Relocate vector table
+    SCB_VTOR = (uint32_t) vector_table;
```
Comment (saleemrashid) [inline util.h]: Should this be masked with `0xFFFF`?

**Comment by saleemrashid** [comment]
Should this be in `util.h`? It now requires `nvic.h` to be generated before compiling `libtrezor.a`

File: util.h [hunk 2]
```diff
@@ -38,4 +41,18 @@ void __attribute__((noreturn)) system_halt(void);
...
+
+    // Set stack pointer
+    __asm__ volatile("msr msp, %0" :: "r" (vector_table->initial_sp_value));
+
+    // Jump to address
+    vector_table->reset();
+
+    // Prevent compiler from generating stack protector code (which causes CPU fault because the stack is moved)
```
Comment (saleemrashid) [inline util.h]: Should this be worded better?

File: util.h (hunk 1 above)
Comment (prusnak) [inline util.h]: No

File: util.h (hunk 2 above)
Comment (prusnak) [inline util.h]: I think it is fine.

**Comment by prusnak** [comment]
Thanks!

**Pull request merged**